db/import.c

```diff
@@ -0,0 +1,1604 @@
+    while (cdb2_next_record(hndl) == CDB2_OK) {
+    }
+
+    logmsg(LOGMSG_DEBUG, "[IMPORT] %s: Flushed source db\n", __func__);
+
+    snprintf(query, sizeof(query),
+             "SELECT filename, content, dir FROM comdb2_files WHERE dir!='tmp' "
+             "AND dir!='savs' ORDER BY filename, offset");
+    rc = cdb2_run_statement(hndl, query);
+
+    if (rc) {
+        logmsg(LOGMSG_ERROR,
+               "[IMPORT] %s: Got an error grabbing files from src db. errstr: "
+               "%s\n",
+               __func__, cdb2_errstr(hndl));
+        rc = 1;
+        goto err;
+    }
+
+    if (gbl_nonames)
+        snprintf(txndir, sizeof(txndir), "%s/logs", gbl_dbdir);
+    else
+        snprintf(txndir, sizeof(txndir), "%s/%s.txn", gbl_dbdir, gbl_dbname);
+
+    while (cdb2_next_record(hndl) == CDB2_OK) {
+        nextFname = (char *)cdb2_column_value(hndl, 0);
+        if (strcmp(nextFname, "checkpoint") == 0) {
+            continue;
+        }
+        int newFile = fname == NULL || strcmp(fname, nextFname) != 0;
+        if (newFile) {
+            if (fname != NULL) {
+                free(fname);
+            }
+            if (f != -1) {
+                close(f);
+            }
+
+            fname = strdup(nextFname);
+
+            char copy_dst[PATH_MAX];
+            const char * dir = (char *)cdb2_column_value(hndl, 2);
+
+            if (strcmp(dir, "") != 0) {
+                snprintf(copy_dst, sizeof(copy_dst), "%s%s%s%s%s", gbl_dbdir,
+                     "/", dir, "/", fname);
+            } else {
+                snprintf(copy_dst, sizeof(copy_dst), "%s%s%s", gbl_dbdir,
+                     "/", fname);
+                
+            }
+
+            f = open(copy_dst, O_WRONLY | O_CREAT | O_APPEND, 0755);
+            if (f == -1) {
+                logmsg(LOGMSG_ERROR,
+                       "[IMPORT] %s: Failed to open file %s (errno: %s)\n",
+                       __func__, copy_dst, strerror(errno));
+                rc = 1;
+                goto err;
+            }
+        }
+        logmsg(LOGMSG_DEBUG, "[IMPORT] %s: Writing chunk to file %s\n",
+               __func__, fname);
+        ssize_t bytes_written = write(f, (char *)cdb2_column_value(hndl, 1),
+                                      cdb2_column_size(hndl, 1));
+        if (bytes_written != cdb2_column_size(hndl, 1)) {
+            logmsg(LOGMSG_ERROR,
+                   "[IMPORT] %s: failed to write to the file (expected: %d "
+                   "got: %ld)\n",
+                   __func__, cdb2_column_size(hndl, 1), bytes_written);
+            rc = 1;
+            goto err;
+        }
+    }
+
+err:
+    if (f != -1) {
+        close(f);
+    }
+
+    if (fname != NULL) {
+        free(fname);
+    }
+
+    return rc;
+}
+
+static int get_my_comdb2_executable(char **p_exe)
+{
+    int rc, size;
+    pid_t pid;
+
+    rc = size = 0;
+    pid = getpid();
+
+#if defined(_LINUX_SOURCE)
+    size = snprintf(NULL, 0, "/proc/%ld/exe", (long) pid);
```

Comment:
This fails under valgrind because the executable is valgrind and not comdb2. I don't know if there are any client use-cases where the executable is not comdb2 -- import would fail for these clients.